src/sif/dynamiccost.cc

```diff
@@ -382,6 +382,9 @@ void ParseBaseCostOptions(const rapidjson::Value& json,
   // shortest
   JSON_PBF_DEFAULT(co, false, json, "/shortest", shortest);
 
+  // TODO: need to check all ranges if they were clamped or not and add a warning if so
+  // thing is, many costing models have another host of custom default ranges
```

Comment:
see if we can pass the warnings from the request pbf to the operator(). also only pointer to "clamped" bool to pass to macro and set warning there (no option name passing).

Reply:
we'd need to pass into `JSON_PBF_RANGED_DEFAULT` the `warnings` part of the request so it can modify it